torchvision/prototype/models/_api.py

```diff
@@ -51,6 +51,19 @@ def verify(cls, obj: Any) -> Any:
             if type(obj) is str:
                 obj = cls.from_str(obj)
             elif not isinstance(obj, cls) and not isinstance(obj, WeightEntry):
+                # Allowing WeightEntry to pass-through is unexpected IMHO.
+                # Since verify() is a factory method, we would expect it to
+                # return only Weights instances, not WeightsEntry instances.
+                # This could lead to bugs if the caller is unaware of that
+                # subtly hidden detail.
+                # From what I understand the motivation for allowing WeightEntry
+                # is for user convenience when calling model builder functions
+                # like resnet18(). IMO for a first version, it's perfectly
+                # reasonable to expect users to write a small wrapper class that
+                # inherits from Weights instead. Maybe this could be re-visited
+                # in the future, but for a first version, I think we should try
+                # to be conservative: if there's a decent way of achieving a
+                # somewhat rare use-case, it's not worth worrying about for now.
```

Comment:
Addressed at commit 21569339074756d877bf527718e54b14eaa4496e of PR #4940

Reply:
Indeed, after discussing with Kai having the ability to pass the path to the pre-trained weights seems to be something they have been traditionally doing (as it also simplifies doing things through configs)

My suggestion was to have a helper function that basically did
```python
def load_state_dict_in_model(model, path):
    state_dict = torch.load(path)
    model.load_state_dict(state_dict)
    return model
```

which I believe covers all use-cases. It adds extra complexity (as it requires one more function call in the config), but I think it might be the best trade-off for explicitly saying that we won't be doing any magic in there.

Reply:
@fmassa are you suggesting to add such a helper in torchvision, or instead to suggest users to implement this themselves?

Reply:
I would like more clarifications on @fmassa's proposal. I wonder if the same thing can be currently achieved by patching the `Weights.get_state_dict()` method, rather than trying to pass callables.

We should note that there are a few places where we need to modify the weights prior loading:
https://github.com/pytorch/vision/blob/d367a01a18a3ae6bee13d8be3b63fd6a581ea46f/torchvision/models/densenet.py#L232-L239